Description: Fix UEFI installation bug
 A previous change in Calamares 3.3.0 caused the UEFI partition to not be
 populated when installing Lubuntu on a UEFI system. This patch reverts the
 change, allowing UEFI installations to work again.
Author: Aaron Rainbolt <arraybolt3@gmail.com>
Origin: upstream
Bug: https://github.com/calamares/calamares/issues/1990
Forwarded: not-needed
Applied-Upstream: https://github.com/calamares/calamares/pull/1991
Last-Update: 2022-06-16
--- a/src/modules/bootloader/main.py
+++ b/src/modules/bootloader/main.py
@@ -798,12 +798,8 @@ def run():
 
     fw_type = libcalamares.globalstorage.value("firmwareType")
 
-    if libcalamares.globalstorage.value("bootLoader") is None:
-        # Don't want a bootloader, but do log that this has an effect:
-        if fw_type != "efi":
-            libcalamares.utils.warning( "Non-EFI system, and no bootloader is set." )
-        else:
-            libcalamares.utils.warning( "EFI system, and no bootloader is set." )
+    if (libcalamares.globalstorage.value("bootLoader") is None and fw_type != "efi"):
+        libcalamares.utils.warning( "Non-EFI system, and no bootloader is set." )
         return None
 
     partitions = libcalamares.globalstorage.value("partitions")
